src/persist/src/file.rs

```diff
@@ -182,6 +183,179 @@ impl Persister for FilePersister {
     }
 }
 
+/// A naive implementation of [Buffer] backed by files.
+pub struct FileBuffer {
+    // Store data and sequence number metadata in two separate files
+    dataz: Arc<Mutex<(File, File)>>,
```

Comment:
this pair of files thing is messy / prone to programmer error - i forget if rust has anonymous structs with named elements

Reply:
looked into this more -- I need the Arc<Mutex.. bit for interior mutability during snapshot, and it doesnt make sense to me to have one file under a arc / mutex and the other one not. if snapshot could take &mut self then i think this struct could just have two files because we already achieve mutual exclusion with the lockfile.

Reply:
doesn't need to be an anonymous struct! just use a non pub one. i've been calling these FooCore (so FileBufferCore), though I'm not tied to that